modules/generic/src/main/scala/pureconfig/generic/EnumerationConfigReaderBuilder.scala

```diff
@@ -13,15 +15,19 @@ import shapeless.labelled._
  * @tparam A the type of objects capable of being read as an enumeration
  */
 trait EnumerationConfigReaderBuilder[A] {
-  def build(transformName: String => String): ConfigReader[A]
+  def build(transformName: String => String, validOptions: Set[String]): ConfigReader[A]
```

Comment:
nit: it looks like at every call this is more like `attemptedOptions` and not always the full set of `validOptions`.

Reply:
To reflect your nittiness back at you, at the time `build` is called, the options haven't been "attempted" yet. I definitely don't have a better suggestion than `attemptedOptions` but it doesn't feel quite right because it gets the temporal ordering wrong. 

Do you have another suggestion.